src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
> we'll have to implement the whole eltype-introspection that Base.map use

I don't think I understand. On a day old master I get this:

```julia
julia> data = Real[3.2]
1-element Array{Real,1}:
 3.2

julia> @code_warntype map(round, data)
Variables:
  #self#::Base.#map
  f::Base.#round
  A::Array{Real,1}

Body:
  begin 
      return $(Expr(:invoke, MethodInstance for _collect(::Array{Real,1}, ::Base.Generator{Array{Real,1},Base.#round}, ::Base.EltypeUnknown, ::Base.HasShape), :(Base._collect), :(A), :($(Expr(:new, :($(QuoteNode(Base.Generator{Array{Real,1},Base.#round}))), :($(QuoteNode(round))), :(A)))), :($(QuoteNode(Base.EltypeUnknown()))), :($(QuoteNode(Base.HasShape())))))
  end::Array{_,1} where _
```

So to my untrained eye it look like map doesn't do better in that regard